markwiki/git/git_integration.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2016, deadc0de6
+'''Versioning of the wiki pages through git'''
+from sh import git
+
+from markwiki.renderer import render_markdown_txt
+
+
+class GitIntegration(object):
+
+    GIT_MSG = 'change on page %s'
+
+    def __init__(self, wiki_path):
+        self.path = wiki_path
+        self.git = git.bake(_cwd=self.path)
+        self.git.init()
+        self._set_config()
+        self._bootstrap()
+
+    def _bootstrap(self):
+        self._add_untracked()
+        self._add_local_change()
+
+    def update_file(self, path):
+        '''called on each update of a wiki page'''
+        self._commit_file(path)
+
+    def _set_config(self):
+        '''set git configs'''
+        self.git.config('--local', 'core.autocrlf', 'input')
+        self.git.config('--local', 'user.name', 'markwiki')
+        self.git.config('--local', 'user.email', '')
+
+    def _add_untracked(self):
+        '''add individual commits for each untracked file'''
+        for f in self.git('ls-files', '-o').split('\n'):
```

Comment:
For these git commands, do you mind using the long flag names instead of the short ones? For example, I think `--others` would be quite a bit more self-documenting than `-o`.